radio/src/thirdparty/Lua/src/liolib.c

```diff
@@ -425,7 +425,7 @@ static int read_chars (lua_State *L, FIL *f, size_t n) {
   luaL_Buffer b;
   luaL_buffinit(L, &b);
   p = luaL_prepbuffsize(&b, n);  /* prepare buffer to read whole block */
-  if (f_read(f, p, LUAL_BUFFERSIZE, &nr) != FR_OK) nr = 0; /* try to read 'n' chars */
+  if (f_read(f, p, n, &nr) != FR_OK) nr = 0; /* try to read 'n' chars */
```

Comment:
do you not want to read the MAX of n and LUAL_BUFFERSIZE to avoid buffer overflows?

Reply:
Don't you mean the *MIN* of the two? 🤭  To prevent overflow you want it to be n, unless n exceeds, at that point it should be constrained to LUAL_BUFFERSIZE . i.e. `if (n > LUAL_BUFFERSIZE) n = LUAL_BUFFERSIZE;`

This might also be a question to put to the Lua developers also, as this is their unmodified code (i.e. c.f. https://www.lua.org/source/5.3/liolib.c.html). Maybe they assumed you won't be using `read_chars()` for long stuff. In our case that would be probably 512 bytes, rather than the 8192 bytes (in some cases) they use have `LUAL_BUFFERSIZE` set to.

Reply:
luaL_prepbuffsize resizes the buffer to the second parameter (n in this case) if needed.
So there is not chance of buffer overflow.

Reply:
Except n still isn't being constrained to LUAL_BUFFERSIZE, though, no? And luaL_prepbuffsize is also checks LUAL_BUFFERSIZE when not explicitly given a upper boundary, and would fail at the `/* not enough space? */` comment. So you can still resize the buffer to larger than the buffer constraint size?

Reply:
Great, sorry for the confusion and yeah, of course MIN! Ha, this is why they don’t let me code any more

Reply:
> 'n' doesn't need to be constrained.

So you are saying n = 784 should be valid input, when LUAL_BUFFERSIZE is 256, rather than be constrained to the configured buffer size?

Reply:
Yes. LUAL_BUFFERSIZE is just a default value that will be used if luaL_prepbuffsize is not called, or n <= LUAL_BUFFERERSIZE.

Reply:
If you look at the comments in the Lua discord channel that started this, he was trying to read nearly 10K.

Reply:
> LUAL_BUFFERSIZE is just a default value that will be used if luaL_prepbuffsize is not called

Fair enough, I was clearly interpreting it wrongly to be the buffer size limit, not just a default. Thank you for clearing that up.